flink-kubernetes/src/main/java/org/apache/flink/kubernetes/highavailability/KubernetesLeaderElectionDriver.java

```diff
@@ -111,7 +117,12 @@ public void close() {
 
         LOG.info("Closing {}.", this);
         leaderElector.stop();
-        kubernetesWatch.close();
+
+        if (kubernetesWatch != null) {
```

Comment:
I think that we should do this check under the `watchLock`. It might be correct now but saying that all accesses to `kubernetesWatch` happen under the lock is easier to understand.